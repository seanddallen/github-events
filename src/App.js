import React, { useState, useEffect, useCallback } from 'react';
import axios from 'axios';
import './App.css';

function App() {

  const [events, setEvents] = useState([])
  const [owner, setOwner] = useState('')
  const [repo, setRepo] = useState('')
  const [eventType, setEventType] = useState('')

  const handleClick = () => {
    axios.get(`https://api.github.com/repos/${owner}/${repo}/events`)
      .then(res => res.data.filter(event => event.type === eventType))
      .then(filteredRes => setEvents(filteredRes))
  }

  return (
    <div className="App" style={{ marginTop: 40, display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
      <div>
        <span>Owner</span><input id="owner" onChange={e => setOwner(e.target.value)} style={{ marginRight: 20 }} />
        <span>Repo</span><input id="repo" onChange={e => setRepo(e.target.value)} style={{ marginRight: 20 }} />
        <span>Event Type</span>
        <select id="eventType" onChange={e => setEventType(e.target.value)}>
          <option>CommitCommentEvent</option>
          <option>CreateEvent</option>
          <option>DeleteEvent</option>
          <option>ForkEvent</option>
          <option>GollumEvent</option>
          <option>IssueCommentEvent</option>
          <option>IssuesEvent</option>
          <option>MemberEvent</option>
          <option>PublicEvent</option>
          <option>PullRequestEvent</option>
          <option>PullRequestReviewCommentEvent</option>
          <option>PushEvent</option>
          <option>ReleaseEvent</option>
          <option>SponsorshipEvent</option>
          <option>WatchEvent</option>
        </select>
        <button id="button" onClick={handleClick}>Search</button>
      </div>

      <div id="events" style={{ marginTop: 20 }}>
        { events.map((event, i) => {
          return (
            <div key={i} style={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-start' }}>
              <div><img src={event.actor.avatar_url} style={{ height: 60, width: 60 }} /></div>
              <div><b>EVENT TYPE: </b>{event.type}</div>
              <div><b>USER: </b>{event.actor.login}</div>
              <div><b>USER ID: </b>{event.actor.avatar_url}</div>
              <div><b>REPO: </b>{event.repo.url}</div>
              <div><b>TIMESTAMP: </b>{event.created_at}</div>
              <hr></hr>
            </div>
          )
        }) }
      </div>
    </div>
  );
}

export default App;
