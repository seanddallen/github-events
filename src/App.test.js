import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />)
});

afterEach(() => {
  jest.clearAllMocks();
});

test('app renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("should match initial layout snapshot", () => {
  expect(wrapper.getElements()).toMatchSnapshot();
});


it('renders an owner input', () => {
  expect(wrapper.find('#owner').length).toEqual(1)
})

it('renders an repo input', () => {
  expect(wrapper.find('#repo').length).toEqual(1)
})

it('renders an event type select dropdown', () => {
  expect(wrapper.find('#eventType').length).toEqual(1)
})

it('should update state of owner input', () => {
  const stateSetter = jest.fn()
  jest.spyOn(React, 'useState')
  //Simulate that mode state value was set to 'new mode value'
  .mockImplementation(stateValue => [stateValue='seantayler', stateSetter])
})

it('should update state of repo input', () => {
  const stateSetter = jest.fn()
  jest.spyOn(React, 'useState')
  //Simulate that mode state value was set to 'new mode value'
  .mockImplementation(stateValue => [stateValue='github-events', stateSetter])
})

it('should list events after button click', () => {
  expect(wrapper.find('#events').length).toBeGreaterThan(0)
})

// it("displays dropdown options", () => {
//   expect(wrapper.find("eventType")).toHaveLength(15);
// });


// it("loads the events", () => {
//   expect(props.fetchEvents).toHaveBeenCalled();
// });

// it('should contain a method called "handleClick"', () => {
//   const wrapper = shallow(<App />)
//   expect(typeof wrapper.instance().handleClick).toBe('function')
// })