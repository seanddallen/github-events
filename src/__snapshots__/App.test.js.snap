// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match initial layout snapshot 1`] = `
Array [
  <div
    className="App"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 40,
      }
    }
  >
    <div>
      <span>
        Owner
      </span>
      <input
        onChange={[Function]}
        style={
          Object {
            "marginRight": 20,
          }
        }
      />
      <span>
        Repo
      </span>
      <input
        onChange={[Function]}
        style={
          Object {
            "marginRight": 20,
          }
        }
      />
      <span>
        Event Type
      </span>
      <select
        onChange={[Function]}
      >
        <option>
          CommitCommentEvent
        </option>
        <option>
          CreateEvent
        </option>
        <option>
          DeleteEvent
        </option>
        <option>
          ForkEvent
        </option>
        <option>
          GollumEvent
        </option>
        <option>
          IssueCommentEvent
        </option>
        <option>
          IssuesEvent
        </option>
        <option>
          MemberEvent
        </option>
        <option>
          PublicEvent
        </option>
        <option>
          PullRequestEvent
        </option>
        <option>
          PullRequestReviewCommentEvent
        </option>
        <option>
          PushEvent
        </option>
        <option>
          ReleaseEvent
        </option>
        <option>
          SponsorshipEvent
        </option>
        <option>
          WatchEvent
        </option>
      </select>
      <button
        onClick={[Function]}
      >
        Search
      </button>
    </div>
    <div
      style={
        Object {
          "marginTop": 20,
        }
      }
    />
  </div>,
]
`;

exports[`should render initial layout 1`] = `
Array [
  <div
    className="App"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 40,
      }
    }
  >
    <div>
      <span>
        Owner
      </span>
      <input
        onChange={[Function]}
        style={
          Object {
            "marginRight": 20,
          }
        }
      />
      <span>
        Repo
      </span>
      <input
        onChange={[Function]}
        style={
          Object {
            "marginRight": 20,
          }
        }
      />
      <span>
        Event Type
      </span>
      <select
        onChange={[Function]}
      >
        <option>
          CommitCommentEvent
        </option>
        <option>
          CreateEvent
        </option>
        <option>
          DeleteEvent
        </option>
        <option>
          ForkEvent
        </option>
        <option>
          GollumEvent
        </option>
        <option>
          IssueCommentEvent
        </option>
        <option>
          IssuesEvent
        </option>
        <option>
          MemberEvent
        </option>
        <option>
          PublicEvent
        </option>
        <option>
          PullRequestEvent
        </option>
        <option>
          PullRequestReviewCommentEvent
        </option>
        <option>
          PushEvent
        </option>
        <option>
          ReleaseEvent
        </option>
        <option>
          SponsorshipEvent
        </option>
        <option>
          WatchEvent
        </option>
      </select>
      <button
        onClick={[Function]}
      >
        Search
      </button>
    </div>
    <div
      style={
        Object {
          "marginTop": 20,
        }
      }
    />
  </div>,
]
`;
